libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -140,6 +175,7 @@ void AP_OSD::update_osd()
 
     update_current_screen();
 
+    screen[current_screen].set_osd(this);
```

Comment:
what we normally do is add an accessor on the backend class. So in this case add get_options() to the AP_OSD_Backend.h